engine/src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -346,34 +384,36 @@ case class JesBackend(actorSystem: ActorSystem)
-    val log = jobLogger(backendCall.jobDescriptor)
-    log.info(s"Call GCS path: ${backendCall.callGcsPath}")
-    val monitoringScript: Option[JesInput] = monitoringIO(backendCall)
+  private def executeOrResume(jobDescriptor: BackendCallJobDescriptor, runIdForResumption: Option[String])(implicit ec: ExecutionContext): Future[ExecutionHandle] = Future {
+    val log = jobLogger(jobDescriptor)
+    log.info(s"Call GCS path: ${jobDescriptor.callGcsPath}")
+    val monitoringScript: Option[JesInput] = jobDescriptor.monitoringIO
     val monitoringOutput = monitoringScript map { _ =>
-      JesFileOutput(s"$MonitoringParamName-out", backendCall.defaultMonitoringOutputPath.toString, Paths.get(JesMonitoringLogFile), backendCall.workingDisk)
+      JesFileOutput(s"$MonitoringParamName-out", jobDescriptor.defaultMonitoringOutputPath.toString, Paths.get(JesMonitoringLogFile), jobDescriptor.workingDisk)
     }
 
-    val jesInputs: Seq[JesInput] = generateJesInputs(backendCall).toSeq ++ monitoringScript :+ backendCall.cmdInput
-    val jesOutputs: Seq[JesFileOutput] = generateJesOutputs(backendCall.jobDescriptor) ++ monitoringOutput
+    val jesInputs: Seq[JesInput] = generateJesInputs(jobDescriptor).toSeq ++ monitoringScript :+ jobDescriptor.cmdInput
+    val jesOutputs: Seq[JesFileOutput] = generateJesOutputs(jobDescriptor) ++ monitoringOutput
 
-    backendCall.instantiateCommand match {
-      case Success(command) => runWithJes(backendCall, command, jesInputs, jesOutputs, runIdForResumption, monitoringScript.isDefined)
+    jobDescriptor.instantiateCommand match {
+      case Success(command) => runWithJes(jobDescriptor, command, jesInputs, jesOutputs, runIdForResumption, monitoringScript.isDefined)
       case Failure(ex: SocketTimeoutException) => throw ex // probably a GCS transient issue, throwing will cause it to be retried
       case Failure(ex) => FailedExecutionHandle(ex)
     }
   }
 
-  def execute(backendCall: BackendCall)(implicit ec: ExecutionContext): Future[ExecutionHandle] = executeOrResume(backendCall, runIdForResumption = None)
+  def execute(jobDescriptor: BackendCallJobDescriptor)(implicit ec: ExecutionContext): Future[ExecutionHandle] = executeOrResume(jobDescriptor, runIdForResumption = None)
 
-  def resume(backendCall: BackendCall, jobKey: JobKey)(implicit ec: ExecutionContext): Future[ExecutionHandle] = {
+  def resume(jobDescriptor: BackendCallJobDescriptor, jobKey: JobKey)(implicit ec: ExecutionContext): Future[ExecutionHandle] = {
     val runId = Option(jobKey) collect { case jesKey: JesJobKey => jesKey.jesRunId }
-    executeOrResume(backendCall, runIdForResumption = runId)
+    executeOrResume(jobDescriptor, runIdForResumption = runId)
   }
 
+  def downloadRcFile(jobDescriptor: BackendCallJobDescriptor) = authenticateAsUser(jobDescriptor.workflowDescriptor) { implicit gcsFs => Try(jobDescriptor.returnCodeGcsPath.toAbsolutePath.contentAsString) }
```

Comment:
This whole `authenticateAsUser implicit gcs blablabla` is totally useless now actually.
I'll make a separate PR to clean that up